Design.md

```diff
@@ -0,0 +1,40 @@
+# Design Document
+
+This document captures the design decisions made after dicussion in issues. When a design issue is closed, the
+conclusion should be summarized here with a link to the issue.
+
+## Disk Layout
+
+Originally discussed in issue #18
```

Comment:
Not sure if it's just a bug in the Rich Diff but it doesn't seem to be linking properly to #18